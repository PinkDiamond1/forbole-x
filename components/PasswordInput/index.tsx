import { FilledTextFieldProps, IconButton, InputAdornment, TextField } from '@material-ui/core'
import HidePasswordIcon from '../../assets/images/icons/icon_hide.svg'
import ShowPasswordIcon from '../../assets/images/icons/icon_show.svg'
import React from 'react'
import useIconProps from '../../misc/useIconProps'

interface PasswordInputProps extends Partial<FilledTextFieldProps> {}

const PasswordInput: React.FC<PasswordInputProps> = (props) => {
  const iconProps = useIconProps()
  const [isShowingPassword, setIsShowingPassword] = React.useState(false)
  return (
    <TextField
      InputProps={{
        disableUnderline: true,
        endAdornment: (
          <InputAdornment position="end">
            <IconButton onClick={() => setIsShowingPassword((prev) => !prev)}>
              {isShowingPassword ? (
                <ShowPasswordIcon {...iconProps} />
              ) : (
                <HidePasswordIcon {...iconProps} />
              )}
            </IconButton>
          </InputAdornment>
        ),
      }}
      fullWidth
      type={isShowingPassword ? 'text' : 'password'}
      variant="filled"
      {...props}
    />
  )
}

export default PasswordInput
