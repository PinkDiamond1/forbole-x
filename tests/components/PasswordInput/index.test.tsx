import React from 'react'
import renderer from 'react-test-renderer'
import PasswordInput from '../../../components/PasswordInput'

describe('component: PasswordInput', () => {
  it('renders correctly', () => {
    const component = renderer.create(<PasswordInput />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly when password is shown', () => {
    const component = renderer.create(<PasswordInput />)
    renderer.act(() => {
      component.root.findByType('button').props.onClick()
    })
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

afterEach(() => {
  jest.clearAllMocks()
})
