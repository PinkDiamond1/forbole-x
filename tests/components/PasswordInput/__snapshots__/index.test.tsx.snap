// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: PasswordInput renders correctly 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
>
  <div
    className="MuiInputBase-root MuiFilledInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiFilledInput-adornedEnd"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputAdornedEnd MuiFilledInput-inputAdornedEnd"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="password"
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-filled MuiInputAdornment-positionEnd"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            data-file-name="SvgIcon_hide"
            fill="#e0e0e0"
            height={16}
            width={16}
          />
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`component: PasswordInput renders correctly when password is shown 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
>
  <div
    className="MuiInputBase-root MuiFilledInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiFilledInput-adornedEnd"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputAdornedEnd MuiFilledInput-inputAdornedEnd"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-filled MuiInputAdornment-positionEnd"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            data-file-name="SvgIcon_show"
            fill="#e0e0e0"
            height={16}
            width={16}
          />
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
